File: Practice/AllDomains/CoreCS/Algorithms/Strings/FunnyString.cs
/*	author:@shivkrthakur  */
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution 
{
    static void Main(String[] args) 
	{
        int T = Convert.ToInt32(Console.ReadLine());
        for(int t = 0; t < T; t++)
        {
            bool Funny = true;
            string s = Console.ReadLine();
            int [] c = s.Select(x => (int)x).ToArray();
            int len = c.Length;
            for(int i = 1, j = len - 2; i < len && j >= 0; i++, j--)
            {
                int eq1 = Math.Abs(c[i] - c[i-1]);
                int eq2 = Math.Abs(c[j] - c[j+1]);
                if(eq1 != eq2) { Funny = false; break; }
            }
            if(Funny) Console.WriteLine(nameof(Funny));
            else Console.WriteLine($"Not {nameof(Funny)}");
        }
    }
}

File: Practice/AllDomains/CoreCS/Algorithms/Strings/FunnyString.js
/*	author:@shivkrthakur  */
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();    
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function funnyString(str){
    // Complete this function
    for(var j = 1, k = str.length - 2; j < str.length; j++, k--){
      if(Math.abs(str.charCodeAt(j) - str.charCodeAt(j-1)) != Math.abs(str.charCodeAt(k) - str.charCodeAt(k+1))){
        return "Not Funny";
      }
    }
    return "Funny";
}

function main() {
    var q = parseInt(readLine());
    for(var a0 = 0; a0 < q; a0++){
        var s = readLine();
        var result = funnyString(s);
        process.stdout.write("" + result + "\n");
    }

}
